### DISCUSSION TITLE
defaultValues not available to nested fields in a fieldArray

### DISCUSSION THREAD

**Discussion by mgerring** [react-hook-form/react-hook-form#5802] [Q&A]
**Describe the bug**
If I have a field array, and the field array renders a custom component that contains one or more fields, and I use useFormContext within that custom component, that component's fields will not have default values set automatically, even though the form correctly handles the default values.

**To Reproduce**
- Create a form wrapped with FormProvider
- Provide default values to useForm
- Create a field array
- Within the fieldArray's fields, render a custom component that wraps one or more form inputs
- Access the form context inside the custom component via useFormContext
- Register the fields in the custom component

**Codesandbox link (Required)**
https://codesandbox.io/s/serene-cartwright-y7s8p?file=/src/App.js

**Expected behavior**
I expect the fields in the custom component to have their default values set

**Desktop (please complete the following information):**

- macOS
- Firefox
- latest stable as of 6/30/2021

**Comment by bluebill1049** [comment]
<img width="1083" alt="Screen Shot 2021-07-01 at 9 25 34 am" src="https://user-images.githubusercontent.com/10513364/124043395-4b443e80-da4e-11eb-9f5a-69d71475b26e.png">

**Reply by mgerring** [reply]
Seems there's been a mistake here, that expression isn't empty when I open the link to the codesandbox:

![Screen Shot 2021-06-30 at 16 31 23](https://user-images.githubusercontent.com/287109/124043904-f5758500-d9c0-11eb-9369-3684240bb382.png)

**Reply by mgerring** [reply]
I included the button for debugging purposes. If you type what should have been the default value into either of the text fields, and click the submit button, the form correctly reports dirtyFields as empty. If you type something other than the default, the form reports the field as dirty.

**Reply by mgerring** [reply]
I would appreciate it if you would re-open the issue, this is definitely a bug and I'm having to code around it right now.

**Comment by bluebill1049** [comment]
issue has been replied: https://github.com/react-hook-form/react-hook-form/issues/5804 this issue has been respl
